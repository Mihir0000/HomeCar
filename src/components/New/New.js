import React, { useState } from 'react';
import { Card, Container } from 'react-bootstrap';

const New = () => {
    const [name, setName] = useState('');
    const [email, setEmail] = useState('');
    const nameChange = (e) => {
        setName(e.target.value);
    };
    const emailChange = (e) => {
        setEmail(e.target.value);
    };
    const submitHandler = () => {
        //   e.preventDefault();
      //   console.log(name, email);
        setEmail('');
        setName('');
    };
    return (
        <Container>
            <Card className="d-flex" style={{ width: '35%', margin: '0 auto' }}>
                <h3 className="d-flex justify-content-center">Contact</h3>

                <form
                    action=""
                    data-testid="valid-form"
                    className="d-flex justify-content-center p-5 w-100 flex-column "
                >
                    <input
                        data-testid="input-name"
                        type="text"
                        placeholder="Name"
                        value={name}
                        onChange={nameChange}
                        required={true}
                    />
                    <input
                        data-testid="input-email"
                        type="email"
                        placeholder="Email"
                        value={email}
                        onChange={emailChange}
                        required={true}
                    />
                    <input type="text" placeholder="Message" />
                    <input
                        type="checkbox"
                        data-testid="input-checkbox-checked"
                    />
                    <button
                        data-testid="submit-btn"
                        type="submit"
                        className="btn"
                        onClick={submitHandler}
                    >
                        Click me
                    </button>
                </form>
            </Card>
        </Container>
    );
};

export default New;
