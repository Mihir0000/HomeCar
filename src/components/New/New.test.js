import React from 'react';
import {  render, screen } from '@testing-library/react';
import New from './New';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
// import jest from 'jest';

test('snapshot testing --new', () => {
    const component = renderer.create(<New />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

// crash test --new
it('render without crashing --new', () => {
    const div = document.createElement('div');
    render(<New />, div);
    ReactDOM.unmountComponentAtNode(div);
});

// it('should handle submit handler', () => {
//     const onClickSpy = jest.fn();
//     const wrapper = shallow(<New onClick={onClickSpy} />);
//     //  render(<New />);
//     //  const submit = screen.getByTestId('submitBtn');
//     //  expect(submit).toHaveBeenCalled()
//     wrapper.find('button').at(0).simulate('click');
//     //  wrapper.find('Button').props().onClick();
//     expect(onClickSpy).toHaveBeenCalled();
// });
// it('Test click Event', () => {
//     const mockCallBack = jest.fn();
//     const button = shallow(<New onClick={mockCallBack}>Click me</New>);
//     button.find('button').simulate('click');
//     expect(mockCallBack.mock.calls.length).toEqual(0);
// });

describe('button test component', () => {
    it('Test click Event', () => {
      const wrapper = shallow(<New/>);
      wrapper.find('button').at(0).simulate('click');
      expect(wrapper.find('button')).toHaveLength(1);
    });
});

// test('btn click event', () => {
//     const btn = screen.getByTestId('submitBtn');
//     fireEvent.click(btn);
//     expect(btn).toBeInTheDocument();
// });

test('when submit btn clicked (name, email) required', () => {
    render(<New />);
    const Email = screen.getByTestId('input-email');
    const Name = screen.getByTestId('input-name');
    expect(Email).toBeRequired();
    expect(Name).toBeRequired();
});

test('check valid form', () => {
    render(<New />);
    const formData = screen.getByTestId('valid-form');
    expect(formData).not.toBeValid();
});
it('input checkbox checked', () => {
    render(<New />);
    const checkBox = screen.getByTestId('input-checkbox-checked');
    expect(checkBox).not.toBeChecked;
});
