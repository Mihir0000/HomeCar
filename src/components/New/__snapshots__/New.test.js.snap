// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing --new 1`] = `
<div
  className="container"
>
  <div
    className="d-flex card"
    style={
      Object {
        "margin": "0 auto",
        "width": "35%",
      }
    }
  >
    <h3
      className="d-flex justify-content-center"
    >
      Contact
    </h3>
    <form
      action=""
      className="d-flex justify-content-center p-5 w-100 flex-column "
      data-testid="valid-form"
    >
      <input
        data-testid="input-name"
        onChange={[Function]}
        placeholder="Name"
        required={true}
        type="text"
        value=""
      />
      <input
        data-testid="input-email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <input
        placeholder="Message"
        type="text"
      />
      <input
        data-testid="input-checkbox-checked"
        type="checkbox"
      />
      <button
        className="btn"
        data-testid="submit-btn"
        onClick={[Function]}
        type="submit"
      >
        Click me
      </button>
    </form>
  </div>
</div>
`;
